tests/integration_test.py

```diff
@@ -0,0 +1,58 @@
+import os
+import json
+import unittest
+
+from unittest.mock import patch, MagicMock
+from difflib import Differ
+
+from main import generate_html
+
+
+class TestMain(unittest.TestCase):
+    """Test case for the main function."""
+
+    def setUp(self):
+        """Load expected and actual HTML outputs."""
+        with open('./tests/expected.html', 'r') as expected_file:
+            self.expected_html = expected_file.read().splitlines()
+
+        # Load sample data from JSON file
+        with open('./tests/sample_data.json', 'r') as json_file:
+            self.sample_data = json.load(json_file)
+
+        self.user_list = ["yankeexe", "Swechhya", "mesaugat"]
```

Comment:
Let's refrain from using actual github usernames in test.

Reply:
@Swechhya Just pushed the changes with dummy names.